packages/fxa-settings/src/pages/Signin/SigninTotpCode/index.tsx

```diff
@@ -51,20 +62,35 @@ const SigninTotpCode = ({
     submitButtonText: 'Confirm',
   };
 
-  const onSubmit = () => {
-    try {
-      // Check authentication code
-      // logViewEvent('flow', `${viewName}.submit`, ENTRYPOINT_REACT);
-      // Check if isForcePasswordChange
-    } catch (e) {
-      // TODO: error handling, error message confirmation
-      // this should probably use auth-errors and message should be displayed in tooltip or banner
+  const onSubmit = async (code: string) => {
+    const result = await submitTotpCode(code);
+
+    setGeneralError('');
+    setCodeErrorMessage('');
+
+    if (result === true) {
+      setSuccess(true);
+      handleNavigation();
+    } else if (result === false) {
+      setSuccess(false);
+      const localizedErrorMessage = getLocalizedErrorMessage(
+        ftlMsgResolver,
+        AuthUiErrors.INVALID_TOTP_CODE
+      );
+      setCodeErrorMessage(localizedErrorMessage);
+    } else {
```

Comment:
Ah, was wondering how a Bool could be neither `true` or `false` but looks like it could be a Boolean or a GQL error object/string.

Reply:
I personally like the `if (error) { // handle all error cases } else { // handle success cases` (or `if (data) {` instead of the `else` if there's data) but 🤷‍♀️ we're still figuring out our preferred patterns with container components.

Reply:
I've changed the response format to be more 'obvious'